logging/src/logging.cc

```diff
@@ -90,6 +104,59 @@ namespace leatherman { namespace logging {
         g_colorize = color_supported(dst);
     }
 
+    void setup_logging(std::string file_name,
+                       std::ios_base::openmode open_mode,
+                       std::string target,
+                       int rotation_size,
+                       int max_size,
+                       int min_free_space,
+                       std::string locale)
+    {
+        // Remove existing sinks before adding a new one
+        auto core = boost::log::core::get();
+        core->remove_all_sinks();
+
+        // Add the text file backend, in order to set file rotation
+        using backend_t = sinks::text_file_backend;
+        boost::shared_ptr<backend_t> backend(
+            new backend_t(
+                keywords::file_name = file_name.data(),     // file name pattern
+                keywords::open_mode = open_mode,            // file open mode
+                keywords::rotation_size = rotation_size,    // triggers rotation, [char]
+                keywords::auto_flush = true));              // flushes at every record
+
+        using sink_t = sinks::synchronous_sink<backend_t>;
+        boost::shared_ptr<sink_t> sink(new sink_t(backend));
+
+        // Configure the file collector; this will limit the size of log files
+        sink->locked_backend()->set_file_collector(sinks::file::make_collector(
+            keywords::target = target.data(),               // dir for rotated files
+            keywords::max_size = max_size,                  // max for stored files, [byte]
+            keywords::min_free_space = min_free_space));    // min drive space, [byte]
+
+        // Scan old files that match the name pattern; consider them already rotated
+        sink->locked_backend()->scan_for_files();
+
+        sink->set_formatter(&color_formatter);
+
+#if (!defined(__sun) && !defined(_AIX)) || !defined(__GNUC__)
```

Comment:
I'm not refactoring the locale imbue() call to avoid having a template dealing with very different types: a std stream and a shared ptr to a boost log sink.